src/components/Reports/GoalCalculator/GoalCalculator.tsx

```diff
@@ -81,29 +79,26 @@ const StyledDrawer = styled(Box, {
   },
 }));
 
-interface GoalCalculatorProps {
-  isNavListOpen: boolean;
-  onNavListToggle: () => void;
-}
-
-export const GoalCalculator: React.FC<GoalCalculatorProps> = ({
-  isNavListOpen,
-  onNavListToggle,
-}) => {
+export const GoalCalculator: React.FC = () => {
   const {
     steps,
     currentStep,
     isDrawerOpen,
     selectedStepId,
-    setRightPanelContent,
+    toggleRightPanel,
     toggleDrawer,
     setDrawerOpen,
     handleStepChange,
     handleContinue,
   } = useGoalCalculator();
   const { t } = useTranslation();
+  const router = useRouter();
   const iconPanelWidth = theme.spacing(5);
 
+  const handleBack = () => {
+    router.back();
```

Comment:
We'll want to push an explicit path to the parent page because `router.back()` won't work if this is the first page that the user opened in this tab.